### ISSUE TITLE
Add support for more data sources

### ISSUE THREAD

**Issue by chubin** [chubin/wttr.in#1025]
## Possible data sources

Possible data sources to be supported:

* OpenWeatherMap
* weather.gov (https://www.weather.gov/documentation/services-web-api)

**Comment by GroovyJeff** [comment]
> One of the better ones imho: https://pirateweather.net

That appears to require an API key? I assume wttr.in only wants to use sources that don't?

**Comment by fallbackerik** [comment]
How about as a next step, we would build little proof of concept (PoC) scripts that all contain the same request details and show how useful outputs are achieved? Could someone document the requirements for such a PoC script? For me, it's for instance not clear yet how we use historical weather data, or if we always use prediction results, or if we do our own predictions in some areas as well.

**Comment by GroovyJeff** [comment]
> Why? We have nothing against using the keys

So the end user would have to enter their own key?

**Comment by GroovyJeff** [comment]
> [@GroovyJeff](https://github.com/GroovyJeff)  Using keys on the server side to access data services is a different matter

OK, but isn't that an issue? If all wttr.in users are using the server's key (which is hidden on the server's side, so not directly available to the user), won't the key's usage be quickly exhausted by the sheer volume of queries? Or am I missing something?

**Comment by fallbackerik** [comment]
@chubin If we follow your approach, I guess a new keyword would need to get added here: https://github.com/chubin/wttr.in/blob/master/lib/parse_query.py#L79-L101

**Comment by fallbackerik** [comment]
> Erik, I agree with you, and yes, it is a big project, but it is quite intresting (from a technical perspective) and also quite promising. It would significantly enhance the data quality and bring wttr.in to the next level.

Yes, so for me a question is how to make a small next step in the right direction. But even if we add the new keyword to the request parser, it might already mean some refactoring of the code base, because it wasn't meant for this kind of parameter in its current form, right?
